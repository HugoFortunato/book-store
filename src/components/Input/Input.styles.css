
label {
  margin-bottom: 8px;

  display: block;

  font-weight: bold;
}

input {
  width: 100%;

  padding: 8px;

  font-size: 14px;
  border-radius: 4px;
  border: 1px solid #ccc;
}
