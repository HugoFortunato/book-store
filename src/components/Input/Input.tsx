import { InputTypes } from '.'
import './Input.styles.css'

export default function Input({
  label,
  placeholder,
  register,
  value,
  type,
  disabled,
  onInputChange,
}: InputTypes) {
  const handleChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
    if (onInputChange) {
      onInputChange(e.target.value)
    }
  }
  return (
    <div className="input-wrapper">
      <label htmlFor="input-id">{label}</label>
      <input
        data-testid="input-id"
        disabled={disabled}
        type={type}
        placeholder={placeholder}
        value={value}
        onChange={handleChange}
        {...(register && register)}
      />
    </div>
  )
}
