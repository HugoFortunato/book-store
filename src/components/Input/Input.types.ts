import { UseFormRegisterReturn } from 'react-hook-form'

export type InputTypes = {
  label?: string
  placeholder: string
  value?: string
  register?: UseFormRegisterReturn
  onInputChange?: (value: string) => void
  type: string
  disabled?: boolean
}
