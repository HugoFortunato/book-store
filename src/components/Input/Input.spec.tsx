import '@testing-library/jest-dom'
import { fireEvent, render } from '@testing-library/react'

import Input from './Input'

describe('<Input />', () => {
  const Test = () => {
    return <Input placeholder="test" value="test" type="text" />
  }

  it('should render a snapshot', () => {
    const { container } = render(<Test />)

    expect(container).toMatchSnapshot()
  })

  it('should render the input with the correct properties', () => {
    const { getByPlaceholderText } = render(<Test />)
    const inputElement = getByPlaceholderText('test')

    expect(inputElement).toBeInTheDocument()
    expect(inputElement).toHaveAttribute('type', 'text')
    expect(inputElement).toHaveValue('test')
  })

  it('should call the onInputChange callback when input changes', () => {
    const mockInputChangeHandler = jest.fn()
    const { getByPlaceholderText } = render(
      <Input
        placeholder="test"
        value="test"
        type="text"
        onInputChange={mockInputChangeHandler}
      />,
    )
    const inputElement = getByPlaceholderText('test')

    fireEvent.change(inputElement, { target: { value: 'new value' } })

    expect(mockInputChangeHandler).toHaveBeenCalledTimes(1)
    expect(mockInputChangeHandler).toHaveBeenCalledWith('new value')
  })

  it('should be disabled when specified', () => {
    const { getByPlaceholderText } = render(
      <Input placeholder="test" value="test" type="text" disabled />,
    )
    const inputElement = getByPlaceholderText('test')

    expect(inputElement).toBeDisabled()
  })
})
