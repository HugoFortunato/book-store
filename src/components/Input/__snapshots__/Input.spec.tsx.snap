// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render a snapshot 1`] = `
<div>
  <div
    class="input-wrapper"
  >
    <label
      for="input-id"
    />
    <input
      data-testid="input-id"
      placeholder="test"
      type="text"
      value="test"
    />
  </div>
</div>
`;
